statsmodels/base/optimizer.py

```diff
@@ -156,11 +155,14 @@ def _fit(self, objective, gradient, start_params, fargs, kwargs,
         extra_fit_funcs = kwargs.setdefault('extra_fit_funcs', dict())
 
         methods = ['newton', 'nm', 'bfgs', 'lbfgs', 'powell', 'cg', 'ncg',
-                'basinhopping']
+                'basinhopping', 'dogleg']
```

Comment:
I answered the first part in main comments.

about newton: the statsmodels newton predates scipy's dogleg and newton by many years. #449
the newer scipy newton style optimizer are one of the prime candidates for adding as default use, are are potentially big improvements compared to our newton.
Again, our newton works reasonably well (and we have more control over some details, like regularization), so we need separate keywords. But I would guess that if the scipy newton methods work as advertised, then we will switch the discrete_models to use those as default.